File: Readio/src/main/java/ninegle/Readio/subscription/repository/SubscriptionRepository.java
package ninegle.Readio.subscription.repository;

import java.util.List;
import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import ninegle.Readio.subscription.domain.Subscription;

@Repository
public interface SubscriptionRepository extends JpaRepository<Subscription, Long> {
	// 사용자 ID로 구독 조회 메서드
	Optional<Subscription> findByUserId(Long userId);

	List<Subscription> findAllByUserId(Long userId);
}

File: Readio/src/main/java/ninegle/Readio/subscription/repository/SubscriptionRepository.kt
package ninegle.Readio.subscription.repository

import ninegle.Readio.subscription.domain.Subscription
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface SubscriptionRepository : JpaRepository<Subscription, Long> {
    // 사용자 ID로 구독 조회 메서드
    fun findByUserId(userId: Long): Subscription?

    fun findAllByUserId(userId: Long): List<Subscription>
}